### ISSUE TITLE
Unable to use stateful pods with PVCs attached

### ISSUE THREAD

**Issue by nkprince007** [Azure/karpenter#65]
### Version

__Karpenter Version:__ latest

__Kubernetes Version:__ v1.27.7

### Expected Behavior

PVC volumes should be attached to karpenter nodes when scheduled

### Actual Behavior

PVC Volumes are not getting attached with the following error.

```
AttachVolume.Attach failed for volume "<pvc-id>" : rpc error: code = Internal desc = Attach volume <pvc-id> to instance <instance-id> failed with not a vmss instance
```

### Steps to Reproduce the Problem

Apply the following manifest with karpenter running.

```yaml
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: task-pv-claim
spec:
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: 1Gi
---
apiVersion: v1
kind: Pod
metadata:
  name: task-pv-pod
spec:
  volumes:
    - name: task-pv-storage
      persistentVolumeClaim:
        claimName: task-pv-claim
  containers:
    - name: task-pv-container
      image: nginx:alpine
      ports:
        - containerPort: 80
          name: "http-server"
      volumeMounts:
        - mountPath: "/usr/share/nginx/html"
          name: task-pv-storage
```

### Resource Specs and Logs

```
{"level":"INFO","time":"2023-12-15T07:04:12.649Z","logger":"controller.nodeclaim.lifecycle","message":"Selected instance type Standard_F2s_v2","commit":"20da22d-dirty","nodeclaim":"general-purpose-twtsc","nodepool":"general-purpose"}
{"level":"INFO","time":"2023-12-15T07:04:12.649Z","logger":"controller.nodeclaim.lifecycle","message":"Resolved image /CommunityGalleries/AKSUbuntu-38d80f77-467a-481f-a8d4-09b6d4220bd2/images/2204gen2containerd/versions/202312.12.0 for instance type Standard_F2s_v2","commit":"20da22d-dirty","nodeclaim":"general-purpose-twtsc","nodepool":"general-purpose"}
{"level":"DEBUG","time":"2023-12-15T07:04:12.650Z","logger":"controller.nodeclaim.lifecycle","message":"Returning 0 IPv4 backend pools: []","commit":"20da22d-dirty","nodeclaim":"general-purpose-twtsc","nodepool":"general-purpose"}
{"level":"DEBUG","time":"2023-12-15T07:04:12.650Z","logger":"controller.nodeclaim.lifecycle","message":"Creating network interface aks-general-purpose-twtsc","commit":"20da22d-dirty","nodeclaim":"general-purpose-twtsc","nodepool":"general-purpose"}
{"level":"DEBUG","time":"2023-12-15T07:04:12.891Z","logger":"controller.nodeclaim.lifecycle","message":"Successfully created network interface: /subscriptions/bc24f077-ad98-4694-a28a-b00a0a614f11/resourceGroups/rg-temporary-common/providers/Microsoft.Network/networkInterfaces/aks-general-purpose-twtsc","commit":"20da22d-dirty","nodeclaim":"general-purpose-twtsc","nodepool":"general-purpose"}
{"level":"DEBUG","time":"2023-12-15T07:04:12.891Z","logger":"controller.nodeclaim.lifecycle","message":"Creating virtual machine aks-general-purpose-twtsc (Standard_F2s_v2)","commit":"20da22d-dirty","nodeclaim":"general-purpose-twtsc","nodepool":"general-purpose"}
{"level":"DEBUG","time":"2023-12-15T07:04:13.016Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:14.017Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:15.018Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:16.019Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:17.019Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:18.020Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:19.021Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:20.022Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:21.023Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:21.979Z","logger":"controller.provisioner","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:22.024Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:23.025Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:24.026Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:25.027Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:26.027Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:27.028Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:28.029Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:29.030Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:30.031Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:31.032Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:31.979Z","logger":"controller.provisioner","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:32.033Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:33.034Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:34.035Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:35.036Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:36.037Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:37.038Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:38.039Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:39.040Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:40.041Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:41.041Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:41.980Z","logger":"controller.provisioner","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:42.042Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:43.043Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:44.044Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:45.044Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:46.045Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:47.046Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:48.047Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:49.048Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:50.049Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:51.050Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:51.981Z","logger":"controller.provisioner","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:52.051Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:53.051Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:54.052Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:55.053Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:56.054Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:57.055Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:58.056Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:59.057Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:04:59.560Z","logger":"controller.nodeclaim.lifecycle","message":"Created virtual machine /subscriptions/bc24f077-ad98-4694-a28a-b00a0a614f11/resourceGroups/rg-temporary-common/providers/Microsoft.Compute/virtualMachines/aks-general-purpose-twtsc","commit":"20da22d-dirty","nodeclaim":"general-purpose-twtsc","nodepool":"general-purpose"}
{"level":"DEBUG","time":"2023-12-15T07:04:59.560Z","logger":"controller.nodeclaim.lifecycle","message":"Creating virtual machine AKS identifying extension for aks-general-purpose-twtsc","commit":"20da22d-dirty","nodeclaim":"general-purpose-twtsc","nodepool":"general-purpose"}
{"level":"DEBUG","time":"2023-12-15T07:05:00.057Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:05:01.059Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:05:01.983Z","logger":"controller.provisioner","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:05:02.059Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:05:03.060Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:05:04.061Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:05:05.062Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:05:06.063Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:05:07.064Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:05:08.065Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:05:09.066Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:05:10.066Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:05:11.067Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:05:11.984Z","logger":"controller.provisioner","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:05:12.067Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:05:13.068Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:05:14.070Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:05:15.070Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:05:16.071Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:05:17.071Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:05:18.072Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:05:19.073Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:05:20.074Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:05:21.075Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:05:22.075Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:05:23.075Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:05:24.076Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:05:25.077Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:05:26.078Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:05:27.079Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:05:28.080Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:05:29.081Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:05:30.082Z","logger":"controller.disruption","message":"waiting on cluster sync","commit":"20da22d-dirty"}
{"level":"DEBUG","time":"2023-12-15T07:05:30.444Z","logger":"controller.nodeclaim.lifecycle","message":"Created  virtual machine AKS identifying extension for aks-general-purpose-twtsc, with an id of /subscriptions/bc24f077-ad98-4694-a28a-b00a0a614f11/resourceGroups/rg-temporary-common/providers/Microsoft.Compute/virtualMachines/aks-general-purpose-twtsc/extensions/computeAksLinuxBilling","commit":"20da22d-dirty","nodeclaim":"general-purpose-twtsc","nodepool":"general-purpose"}
{"level":"INFO","time":"2023-12-15T07:05:30.444Z","logger":"controller.nodeclaim.lifecycle","message":"launched new instance","commit":"20da22d-dirty","nodeclaim":"general-purpose-twtsc","nodepool":"general-purpose","launched-instance":"/subscriptions/bc24f077-ad98-4694-a28a-b00a0a614f11/resourceGroups/rg-temporary-common/providers/Microsoft.Compute/virtualMachines/aks-general-purpose-twtsc","hostname":"aks-general-purpose-twtsc","type":"Standard_F2s_v2","zone":"3","capacity-type":"on-demand"}
{"level":"INFO","time":"2023-12-15T07:05:30.444Z","logger":"controller.nodeclaim.lifecycle","message":"launched nodeclaim","commit":"20da22d-dirty","nodeclaim":"general-purpose-twtsc","nodepool":"general-purpose","provider-id":"azure:///subscriptions/bc24f077-ad98-4694-a28a-b00a0a614f11/resourceGroups/rg-temporary-common/providers/Microsoft.Compute/virtualMachines/aks-general-purpose-twtsc","instance-type":"Standard_F2s_v2","zone":"","capacity-type":"on-demand","allocatable":{"cpu":"1900m","ephemeral-storage":"128G","memory":"2014Mi","pods":"110"}}
{"level":"DEBUG","time":"2023-12-15T07:05:30.466Z","logger":"controller.nodeclaim.lifecycle","message":"registered nodeclaim","commit":"20da22d-dirty","nodeclaim":"general-purpose-twtsc","nodepool":"general-purpose","provider-id":"azure:///subscriptions/bc24f077-ad98-4694-a28a-b00a0a614f11/resourceGroups/rg-temporary-common/providers/Microsoft.Compute/virtualMachines/aks-general-purpose-twtsc","node":"aks-general-purpose-twtsc"}
{"level":"INFO","time":"2023-12-15T07:05:30.487Z","logger":"controller.nodeclaim.lifecycle","message":"initialized nodeclaim","commit":"20da22d-dirty","nodeclaim":"general-purpose-twtsc","nodepool":"general-purpose","provider-id":"azure:///subscriptions/bc24f077-ad98-4694-a28a-b00a0a614f11/resourceGroups/rg-temporary-common/providers/Microsoft.Compute/virtualMachines/aks-general-purpose-twtsc","node":"aks-general-purpose-twtsc"}
{"level":"DEBUG","time":"2023-12-15T07:05:31.354Z","logger":"controller.disruption","message":"adding requirements derived from pod volumes, [{topology.disk.csi.azure.com/zone In [eastus-3]}]","commit":"20da22d-dirty","pod":"default/task-pv-pod"}
{"level":"DEBUG","time":"2023-12-15T07:09:04.886Z","logger":"controller.disruption","message":"adding requirements derived from pod volumes, [{topology.disk.csi.azure.com/zone In [eastus-3]}]","commit":"20da22d-dirty","pod":"default/task-pv-pod"}
```

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by nkprince007** [comment]
@Bryce-Soghigian @tallaxes can you please look into this and let me know if there's any misconfiguration?

**Comment by Bryce-Soghigian** [comment]
@nkprince007 Found the root cause, investigating the best fix. Based on what we see, kubernetes version 1.28 should not have this problem, could you please give it a try?

Specifying 1.28, you should see pvcs getting attached successfully. 
```
Normal   SuccessfulAttachVolume  48m   attachdetach-controller  AttachVolume.Attach succeeded for volume "pvc-<guid-redacted>"
```

**Comment by nkprince007** [comment]
Can you give us a timeline fpr the fix, please? @Bryce-Soghigian

**Comment by nkprince007** [comment]
@tallaxes @Bryce-Soghigian any update on this issue?

**Comment by Bryce-Soghigian** [comment]
@nkprince007  I am unable to reproduce for 1.28+ for default nap settings, keep in mind we haven't done any private cluster testing. 


This is the cluster configuration I am using
```
	az aks create          --name $(AZURE_CLUSTER_NAME) --resource-group $(AZURE_RESOURCE_GROUP) --attach-acr $(AZURE_ACR_NAME) \
		--enable-managed-identity --node-count 3 --generate-ssh-keys -o none --network-dataplane cilium --network-plugin azure --network-plugin-mode overlay -k 1.28.3
```

Can you share your configuration as well? That way I can create a cluster with your configuration and attempt to repro the issue? 


See my working example here
```yaml
~/dev/karpenter (main*) » k describe pod task-pv-pod                                                                                                                                                                                                            sillygoose@Bryces-MacBook-Pro
Name:             task-pv-pod
Namespace:        default
Priority:         0
Service Account:  default
Node:             aks-general-purpose-lqztt/10.224.0.7
Start Time:       Thu, 21 Dec 2023 20:43:23 -0800
Labels:           <none>
Annotations:      <none>
Status:           Running
IP:               10.244.3.172
IPs:
  IP:  10.244.3.172
Containers:
  task-pv-container:
    Container ID:   containerd://8169b529769b580614f88d3181db0a2ac68efd1537e81e6e90741feb3201c0bd
    Image:          nginx:alpine
    Image ID:       docker.io/library/nginx@sha256:a59278fd22a9d411121e190b8cec8aa57b306aa3332459197777583beb728f59
    Port:           80/TCP
    Host Port:      0/TCP
    State:          Running
      Started:      Thu, 21 Dec 2023 20:43:42 -0800
    Ready:          True
    Restart Count:  0
    Environment:    <none>
    Mounts:
      /usr/share/nginx/html from task-pv-storage (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-4pd5f (ro)
Conditions:
  Type              Status
  Initialized       True
  Ready             True
  ContainersReady   True
  PodScheduled      True
Volumes:
  task-pv-storage:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  task-pv-claim
    ReadOnly:   false
  kube-api-access-4pd5f:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   BestEffort
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type     Reason                  Age   From                     Message
  ----     ------                  ----  ----                     -------
  Warning  FailedScheduling        117s  default-scheduler        0/3 nodes are available: 3 node(s) had untolerated taint {CriticalAddonsOnly: true}. preemption: 0/3 nodes are available: 3 Preemption is not helpful for scheduling..
  Normal   Scheduled               56s   default-scheduler        Successfully assigned default/task-pv-pod to aks-general-purpose-lqztt
  Normal   Nominated               116s  karpenter                Pod should schedule on: nodeclaim/general-purpose-lqztt
  Normal   SuccessfulAttachVolume  47s   attachdetach-controller  AttachVolume.Attach succeeded for volume "pvc-<redacted>"
  Normal   Pulling                 40s   kubelet                  Pulling image "nginx:alpine"
  Normal   Pulled                  38s   kubelet                  Successfully pulled image "nginx:alpine" in 2.746s (2.746s including waiting)
  Normal   Created                 38s   kubelet                  Created container task-pv-container
  Normal   Started                 38s   kubelet                  Started container task-pv-container
  
  ```

**Comment by nkprince007** [comment]
Unfortunately @Bryce-Soghigian, the cluster is not created through CLI, hence I can't give you the exact command that we ran using azure cli.

However, we do have the JSON represenation of the cluster configuration from Azure. I'm attaching it below. (We have redacted PII for compliance reasons, but other than that all the information should be available)

[cluster.json](https://github.com/Azure/karpenter/files/13751499/cluster.json)
